### ISSUE TITLE
postrender event for decluttered items

### ISSUE THREAD

**Issue by mike-000** [openlayers/openlayers#11606]
**Is your feature request related to a problem? Please describe.**

Examples which use postrender events do not work correctly with decluttering as the decluttered items are rendered after the layer's postrender event

https://codesandbox.io/s/trusting-edison-vnzsk?file=/main.js

A workaround is to postpone the process with a short timeout, but in some cases such as rapid panning or a fast moving layer slice the next prerender event may fire before the timeout, requiring an extra check

https://codesandbox.io/s/sleepy-williams-msqeh?file=/main.js

**Describe the solution you'd like**
It would be much more convenient if the map fired a postrender event (which would need to include the layer className to identify which layer(s) it applied to) after decluttered items are rendered..

**Comment by ahocevar** [comment]
Maybe a `targetrender` event on the map, with the target element (canvas) instead of the className as event property? It would be dispatched when all rendering on a render target is done for the current render cycle.

For the existing `postrender` event on the layer, it should be enough to document that decluttered items will be rendered after that event.

Such a change would also be easier after #11592 is merged, because of the new `renderDeclutter` logic which is then tied to layers instead of the map.
